models/transaction.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * This file contains wrapper functions to interact with the DB.
+ * This will contain the DB schema if we start consuming an ORM for managing the DB operations
+ */
+
+const firestore = require('../utils/firestore')
+const transactionsModel = firestore.collection('transaction')
+
+/**
+ * fetch latest N transactions from transactions collection for specific userid provided
+ *
+ * @param userId { String }: User Id String to be used to fetch latest transactions
+ * @param limit { number }: no of records to fetch from transaction table, if not specified in URL then default will be 10
+ * @param offset { number }: starting index of set of records, default value is 0, useful for pagination
+ * @param orderBy { String }: to order the transactions in ascending or descending manner, by default will fetch latest first
+ * @returns {Promise<{transactions: []}>}
+ */
+const fetchTransactionsByUserId = async (userId, offset, limit, orderBy) => {
+  const transactionsRef = await transactionsModel.where('userId', '==', userId).get()
+  const transactions = []
+  transactionsRef.forEach((doc) => {
+    const transaction = doc.data()
```

Comment:
Yes, our test cases fail.

Reply:
No, what I meant is, if I pass `bobTheBuilder` in `userId` which doesn't exist in DB, does the API break with 5xx error code? Are we handling this case?

Reply:
I guess the API will break in that case. You should have a check for doc to exist, check how other model functions are handling this case

Reply:
If the doc doesn't exist, it won't have `.data()` and the control would go directly to catch handler and API would return 5xx, not ideal case
Firestore does provide a `.exists` or some other flag

Reply:
> No, what I meant is, if I pass `bobTheBuilder` in `userId` which doesn't exist in DB, does the API break with 5xx error code? Are we handling this case?

But the userID is passed as per who is logged in.

Reply:
Yeah, got your point. Working on it.

Reply:
Is this solved?

Reply:
yeah, So the `if` statement checks with is there any data returned by the `where` query or not. 
So if the userID is wrong then it will not return anything and log error.

Reply:
> So if the userID is wrong then it will not return anything and log error.

This is bad UX. We're giving 5xx even when we know the reason. We should be giving 4xx here
We should be handling the cases which we know can happen. If we know that passing a non-existing username can break the API, we should handle it.

Reply:
Done